cpp/src/parquet/column_reader.h

```diff
@@ -339,27 +347,49 @@ class RecordReader {
   bool read_dictionary() const { return read_dictionary_; }
 
  protected:
+   // Indicates if we can have nullable values.
   bool nullable_values_;
 
   bool at_record_start_;
   int64_t records_read_;
 
+  // Stores values. These values are populated based on each ReadRecords call.
+  // No extra values are buffered for the next call. SkipRecords will not
+  // add any value to this buffer.
+  std::shared_ptr<::arrow::ResizableBuffer> values_;
+  // In the case of false (BYTE_ARRAY), don't allocate the values buffer
+  // (when we directly read into builder classes).
+  bool uses_values_;
```

Comment:
why was this moved up?

Reply:
I must have re-ordered it to improve readability.